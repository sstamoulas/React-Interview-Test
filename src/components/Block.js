import React from "react";
import PropTypes from "prop-types";
import {
  Box,
  makeStyles,
} from "@material-ui/core";
import colors from "../constants/colors";

const Block = ({ block }) => {
  const classes = useStyles();

  let index = block.attributes.index;
  index = index < 10 ? `00${index}` : index < 100 ? `0${index}` : `${index}`;

  return (
    <Box className={classes.root}>
      <div className={classes.index}>{index}</div>
      <div className={classes.content}>{block.attributes.data}</div>
    </Box>
  );
};

const useStyles = makeStyles((theme) => ({
  root: {
    backgroundColor: colors.grey,
    margin: "4px",
  },
  content: {
    color: colors.text,
    fontSize: theme.typography.pxToRem(13),
    fontWeight: "500",
    padding: "0 0 5px 7px",
  },
  index: {
    color: "blue",
    fontSize: theme.typography.pxToRem(11),
    padding: "7px 0 0 8px",
  }
}));

Block.propTypes = {
  block: PropTypes.object.isRequired,
};

export default Block;
