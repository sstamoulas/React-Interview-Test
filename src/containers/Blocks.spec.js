import React from "react";
import { shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import thunk from 'redux-thunk';
import { Provider } from "react-redux";
import { create } from "react-test-renderer";

import ConnectedBlocks, { Blocks } from "./Blocks";
import Block from "./../components/Block";

describe("<Blocks />", () => {
  let wrapper;

  const node = {      
    url: 'https://thawing-springs-53971.herokuapp.com',
    online: false,
    name: 'Node 1',
    loading: false,
    blocks: [{ 
      "id": "5",
      "type": "blocks",
      "attributes": {
        "index": 1,
        "timestamp": 1530679678,
        "data": "The Human Torch",
        "previous-hash": "KsmmdGrKVDr43/OYlM/oFzr7oh6wHG+uM9UpRyIoVe8=",
        "hash": "oHkxOJWOKy02vA9r4iRHVqTgqT+Afc6OYFcNYzyhGEc="
      }
    },
    {
      "id": "6",
      "type": "blocks",
      "attributes": {
        "index": 2,
        "timestamp": 1530679684,
        "data": "is denied",
        "previous-hash": "oHkxOJWOKy02vA9r4iRHVqTgqT+Afc6OYFcNYzyhGEc=",
        "hash": "9xr57PW8RVzeOniNP2lPVzAOu97x12mpDgjv70z5vo4="
      }
    },
    {
      "id": "7",
      "type": "blocks",
      "attributes": {
        "index": 3,
        "timestamp": 1530679689,
        "data": "a bank loan",
        "previous-hash": "9xr57PW8RVzeOniNP2lPVzAOu97x12mpDgjv70z5vo4=",
        "hash": "YGLfNDMC2x2m5kwb3q+Ne/uCL4sFUnX/sQwzuwijx8A="
      }
    }],
  };

  it("should contain <Block />", () => {
    const wrapper = shallow(
      <Blocks
        node={node}
      />
    );

    expect(wrapper.find(Block).length).toEqual(3);
  });

  it("should match snapshot", () => {
    const middlewares = [thunk];
    const store = configureMockStore(middlewares)({node});
    const component = create(
      <Provider store={store}>
        <ConnectedBlocks node={node} />
      </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
