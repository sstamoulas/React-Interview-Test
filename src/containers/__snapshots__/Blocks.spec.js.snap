// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blocks /> should match snapshot 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-4 makeStyles-root-1"
  >
    <div
      className="makeStyles-index-3"
    >
      001
    </div>
    <div
      className="makeStyles-content-2"
    >
      The Human Torch
    </div>
  </div>,
  <div
    className="MuiBox-root MuiBox-root-5 makeStyles-root-1"
  >
    <div
      className="makeStyles-index-3"
    >
      002
    </div>
    <div
      className="makeStyles-content-2"
    >
      is denied
    </div>
  </div>,
  <div
    className="MuiBox-root MuiBox-root-6 makeStyles-root-1"
  >
    <div
      className="makeStyles-index-3"
    >
      003
    </div>
    <div
      className="makeStyles-content-2"
    >
      a bank loan
    </div>
  </div>,
]
`;
